timer/base_timer.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import (
+	"errors"
+	"time"
+)
+
+// baseTimer is the type used for all the timers that won't have children
+// It's implementation is more lightweight than nodeTimer and can be used as a standalone timer using NewTimer
+type baseTimer struct {
+	timeCache
+
+	// start is the time when the timer was started
+	start time.Time
+
+	// spent is the time spent on the timer, set after calling stop
+	spent time.Duration
+
+	// config is the configuration of the timer
+	config Config
+
+	// parentComponent is the component stored in the map of the parent timer. We keep a reference to it to update its spent time
+	// once the timer is stopped. It may be nil if the timer is not a child of a tree timer, like when calling NewTimer.
+	parentComponent *component
```

Comment:
(Nano-optimization) Suggest running `betteralign`... In particular, there is positive impact on time spent by GC on types if you keep all pointers at the start of the structure.

Reply:
Benchmarks says we won ~5% perfs. Commiting this :+1: